controllers/spannerautoscaleschedule_controller.go

```diff
@@ -86,29 +92,64 @@ func (r *SpannerAutoscaleScheduleReconciler) Reconcile(ctx context.Context, req
 		return ctrlreconcile.Result{}, nil
 	}
 
+	sasFinalizerName := "spannerautoscaleschedule.spanner.mercari.com/finalizer"
+
+	// register our finalizer if the object is not being deleted
+	if sas.ObjectMeta.DeletionTimestamp.IsZero() && !ctrlutil.ContainsFinalizer(&sas, sasFinalizerName) {
+		log.V(1).Info("adding finalizer to schedule", "schedule", sas, "finalizer", sasFinalizerName)
+		ctrlutil.AddFinalizer(&sas, sasFinalizerName)
+		if err := r.ctrlClient.Update(ctx, &sas); err != nil {
+			return ctrl.Result{}, err
```

Comment:
It seems better to output the log.
Like this https://github.com/mercari/spanner-autoscaler/blob/master/controllers/spannerautoscaler_controller.go#L284-L285

Reply:
Thanks, fixed!